### DISCUSSION TITLE
Share your feedback with NiChart. It's quick and easy!

### DISCUSSION THREAD

**Discussion by gurayerus** [CBICA/NiChart_Project#72] [Announcements]
Hello,

The feedback from our community is incredibly valuable to NiChart. We would be grateful if you could take a few minutes to share your thoughts by completing our surveys.

* [NiChart User Experience](https://forms.office.com/r/mM1kx1XsgS)

* [Shaping the Future of NiChart](https://forms.office.com/r/acwgn2WCc4)

You can view our quick-start cloud demo videos in our YouTube channel

* [NiChart Demo1](https://www.youtube.com/watch?v=RrnibCrNiHA)  

* [NiChart Demo2](https://www.youtube.com/watch?v=aw5AK0UQtgo)

Visit our [web page](https://neuroimagingchart.com) for an overview of NiChart or simply run it on [our cloud app](https://cloud.neuroimagingchart.com)

Thank you!

**Comment by A-Cronnex** [comment]
Hello, my name is Aaron. I am future researcher from the Univeristy of Panama. Two of the members of your project have personally recommended the usage of your pipeline, I have talked with two of them in the NITRC forum. However, I cannot use the tool in the same way as the tutorial shows because I can't choose the output directory based on my own data and neither I can choose the type of training I wanna do (In the DLMUSE part, I can't chose between cuda and the others) I am very eager to see if your tool can help me with my research project, but I don't know why it's missing such key options that appear in the tutorial. 

Edit: I will share this comment also in the Software discussion part

**Comment by AlexanderGetka-cbica** [comment]
Hi Aaron,

Thank you for your interest in NiChart. I'm happy to help troubleshoot.

Are you using the cloud portal or the local version of the application?

On the cloud, we don't allow users to select CUDA/MPS/etc. This option is only meant for local usage in case one has a Mac, doesn't have an Nvidia GPU, and so on (by default on cloud we always use CUDA). If this option isn't showing up on local, this may be a bug.

Also on the cloud, output is automatically stored. To get results you can just click download at the end of each step (for instance, to download DLMUSE ROIs, scroll to the bottom after DLMUSE finishes, click download and select scans.) On the local version, you should be able to select  the output folder, but if not you can check under the "output_folder" subdirectory in the cloned repo.

For clarity, we don't currently offer training via this interface, only inference of our pretrained models on submitted data.

I hope this is helpful, and please let me know if you have any questions.

Best regards,
Alex Getka

Lead Application Developer
Center for Biomedical Computing and Analytics
University of Pennsylvania



________________________________
From: Aaron C. ***@***.***>
Sent: Thursday, January 23, 2025 4:34 PM
To: CBICA/NiChart_Project ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [External] Re: [CBICA/NiChart_Project] Share your feedback with NiChart. It's quick and easy! (Discussion #72)


Hello, my name is Aaron. I am future researcher from the Univeristy of Panama. Two of the members of your project have personally recommended the usage of your pipeline, I have talked with two of them in the NITRC forum. However, I cannot use the tool in the same way as the tutorial shows because I can't choose the output directory based on my own data and neither I can choose the type of training I wanna do (In the DLMUSE part, I can't chose between cuda and the others) I am very eager to see if your tool can help me with my research project, but I don't know why it's missing such key options that appear in the tutorial.

**Reply by A-Cronnex** [reply]
Hi Alex, thank you! I appreciate your responde a lot

I am using the cloud version on a computer with Ubuntu. All the images are already in a Nifti format, so I am using directly the DLMUSE method.

I have already put the images under the DLMUSE. Is it taking quite a while, there are 39 images in total. Is it normal that it takes a lot of time, or I did something wrong in the process?

Edit: It worked correctly after a while of waiting

**Comment by AlexanderGetka-cbica** [comment]
Hi Aaron,

Under normal conditions, in my observation, it typically takes perhaps around 20 to 40 seconds per T1 scan for the full DLMUSE pipeline. (We are always striving toward lowering this number and increasing parallel processing even further, but this is a complex process even on the cloud. Our first priority right now are changes that should speed up the process by a factor of approximately 10x or more, and this should be available within the next few months.)

 Looking through the logs right now, I see multiple submissions of 160+ scans — it seems the platform is a little more popular than we expected for this at the current moment, and this might be causing some issues with the processing speed due to high GPU usage.

If your process hasn't finished, it may be best to try again later. Also note that, at least for up to 24 hours, you can simply let the process run and then download the results later.

For example, if i create a study "MyFirstStudy",  upload my files there, then run DLMUSE, I don't have to even stay on the page. If I go back to the same page later and select MyFirstStudy again, and results are available from the first run, it should still be able to be downloaded at the bottom. (However, this will not work if your results have been automatically cleaned up, which occurs after some time, or if the original run never finished).

Hope this helps and I hope to get back to you soon regarding updates.

Thanks,
Alex


________________________________
From: Aaron C. ***@***.***>
Sent: Thursday, January 23, 2025 7:53 PM
To: CBICA/NiChart_Project ***@***.***>
Cc: Getka, Alexander ***@***.***>; Comment ***@***.***>
Subject: [External] Re: [CBICA/NiChart_Project] Share your feedback with NiChart. It's quick and easy! (Discussion #72)


Hi Alex, thank you! I appreciate your responde a lot

I am using the cloud version on a computer with Ubuntu. All the images are already in a Nifti format, so I am using directly the DLMUSE method.

I have already put the images under the DLMUSE. Is it taking quite a while, there are 39 images in total. Is it normal that it takes a lot of time, or I did something wrong in the process?

**Comment by A-Cronnex** [comment]
Hello NiChart team. This is Aaron again. My team and I have been using your tool for our research project, however we have stumbled with a couple of issues that make us unable to further extract the ROIs from our data. These are the errors, in particular: 
![Screenshot from 2025-04-28 22-42-34](https://github.com/user-attachments/assets/15f59a17-257e-4ef3-a9d6-625037391198)
![Screenshot from 2025-04-28 22-52-50](https://github.com/user-attachments/assets/47308101-22bc-4cba-8eb8-6dfd5b34cb9f)
![Screenshot from 2025-04-29 06-08-44](https://github.com/user-attachments/assets/84d967a8-41d4-4383-b8db-21c1e3ef3a8f)
![Screenshot from 2025-04-29 06-31-44](https://github.com/user-attachments/assets/5b8ca990-f285-4cc5-8d66-a82185b16ff3)

I don't really know a way in which this can be solved

**Reply by spirosmaggioros** [reply]
@AlexanderGetka-cbica

**Reply by AlexanderGetka-cbica** [reply]
Hi Aaron, I believe I found a solution. It will take some time to deploy, but if you're willing, it would be super helpful to coordinate with you to see if things are working as expected afterward.

**Reply by A-Cronnex** [reply]
Sure! I am delighted to. Tell the moment you can, I will arrive to my home from college around 2:30 pm, Panamanian hours

**Comment by AlexanderGetka-cbica** [comment]
Hi @A-Cronnex , thanks for letting us know. I am in the process of fixing some cloud-specific functionality we recently deployed -- there should be a warning on the homepage about this. I'll be debugging this today and will get back to you ASAP.

**Comment by A-Cronnex** [comment]
![Screenshot from 2025-04-29 21-28-02](https://github.com/user-attachments/assets/b4e462fe-b835-4bea-bee4-2d03964d6682)
Hello Alexander. I'd like to report that I ams till getting this error. I am confused in some aspects of this, for example, does that happen because the platform gets overloaded with nifti files? All the files there weight in total 3.9 GB+

Edits: I am still getting the error and it's strange because sometimes it ocasionally outputs the information. Do I have to take into consideration which time of the day I do the extractions? Neither my team nor me could get the DLMUSE csv when we did the extractions at night. However, when I did it in the day I managed to get some .nii images extracted, which can be shown in the second image

![Screenshot from 2025-04-29 22-20-53](https://github.com/user-attachments/assets/81fbb295-0e30-4ac8-8df0-4a2a1508a1b0)
![Screenshot from 2025-04-29 22-22-44](https://github.com/user-attachments/assets/28647624-174c-49a9-980c-a57a8fc02520)

**Comment by A-Cronnex** [comment]
One of my colleague has reported the same error. He stated that he was able to extract the ROIs from 10 to 20, which ended up getting the same error showed above. He decided to try with 5 nifti files on each extractions, in which he was able to extract them. However, durign the extraction one of them got stuck. Is it because of a server overload?![IMG-20250430-WA0001(1).jpg](https://github.com/user-attachments/assets/906be7c7-23b3-475c-945e-0f94a9a8595f)

![IMG-20250430-WA0002.jpg](https://github.com/user-attachments/assets/502b3d01-fe76-431b-be38-8c1df4ad9659)

**Reply by AlexanderGetka-cbica** [reply]
There are some issues with the backend infrastructure that are causing these problems right now. I'm rolling back right now and will update when things seem to work properly again.

**Reply by AlexanderGetka-cbica** [reply]
Hi Aaron, some updates and clarification:

First, I notice from your original error messages that there are actually multiple errors being encountered:
 1.  the defaced images are not being found. Those files don't even appear to be present in the back end when I checked, so this is probably an application bug in how we detect the common prefix/suffix for the files. As a workaround in the mean time, this might be fixed by renaming to a subject ID more standard to the rest.
 2. At some point, the application was writing the files successfully in earlier steps, then failing when a later step tried to read those same files. This was a result of the new backend infrastructure using a networked filesystem, which strangely does not have consistency guarantees even on the same device. This is why your results were sometimes succeeding, sometimes failing. But I believe I have fixed this by forcing the filesystem to synchronize all writes immediately. This should guarantee consistency and prevent the "FileNotFoundError" you were seeing. **Please let me know** if this keeps occurring!
 3. Regarding the DLMUSE stuck on "wait for it...", I was experiencing the same issue, but this appears to be fixed now. A server overload scenario could potentially occur, but in this case it seems like the instances were just in an error state and being unresponsive.

**Comment by A-Cronnex** [comment]
Hello NiChart team, this is Aaron again.

My colleagues and I are still encountering the same error. Though I was able to extract all files for my class, they couldn't do the same for their classes. Their classes come from the same dataset. The name changing process was done and the same error still appears. 
![Screenshot from 2025-05-10 21-17-02](https://github.com/user-attachments/assets/3197f99a-dcb8-4ec7-9aaa-e74f787963ec)
![Screenshot from 2025-05-10 21-58-44](https://github.com/user-attachments/assets/acefbf31-78a2-407c-9b36-a79cd2204617)
![Report_1](https://github.com/user-attachments/assets/72c0ca6b-beca-443c-8a0c-93e98f795f21)

**Comment by A-Cronnex** [comment]
@AlexanderGetka-cbica  Hello Alexander. I am encountering another bug in the app. After extracting the files, I get the same error of "file name not found" but on top of that, after I reload the webpage, I get yet another error. This is the output message of it:

![Screenshot from 2025-05-18 17-56-45](https://github.com/user-attachments/assets/b1f85574-7952-4279-a7b9-4dfe91da25d6)

This has been particulary problematic for me because I have to extract those nifti files very soon for our project, and those are the only ones that I got left.
